server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -169,6 +171,22 @@ public static IndexRequest getIndexWriteRequest(DocWriteRequest<?> docWriteReque
         return indexRequest;
     }
 
+    public static <Response extends ReplicationResponse & WriteResponse> ActionListener<BulkResponse> wrapBulkResponseAsSingle(
```

Comment:
Naming nit: this is more like unwrapping than wrapping? How about something like this?

```suggestion
    public static <Response extends ReplicationResponse & WriteResponse> ActionListener<BulkResponse> unwrappingSingleItemBulkResponse(
```

Reply:
I'm good with unwrapping. I went with wrap because under the hood we call `ActionListener.wrap(...)` so it's wrapping in a different sense. 

Is `unwrapAs` instead of `unwrapping` okay with you?